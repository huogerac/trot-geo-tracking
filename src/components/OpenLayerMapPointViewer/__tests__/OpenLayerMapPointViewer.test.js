import { OpenLayerMapPointViewer } from ".."
import { mount } from "@vue/test-utils"

const createComponent = (options) => mount(OpenLayerMapPointViewer, options)

describe("OpenLayerMapPointViewer", () => {
  test("Render correctly", () => {
    const wrapper = createComponent()

    expect(wrapper.html()).toMatchSnapshot()
  })
  test("Render correctly - with the passed props", () => {
    const wrapper = createComponent({
      props: {
        positionsList: ["35.7040744", "139.5577317"],
        center: ["35.7040744", "139.5577317"],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
