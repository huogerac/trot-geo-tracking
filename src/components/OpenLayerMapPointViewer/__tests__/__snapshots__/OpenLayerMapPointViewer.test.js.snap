// Vitest Snapshot v1

exports[`OpenLayerMapPointViewer > Render correctly - with the passed props 1`] = `
"<ol-map load-tiles-while-animating=\\"true\\" load-tiles-while-interacting=\\"true\\" style=\\"height: 400px;\\">
  <ol-view center=\\"35.7040744,139.5577317\\" rotation=\\"0\\" zoom=\\"17\\" projection=\\"EPSG:4326\\"></ol-view>
  <ol-tile-layer>
    <ol-source-osm></ol-source-osm>
  </ol-tile-layer>
  <ol-vector-layer>
    <ol-source-vector>
      <ol-feature>
        <ol-geom-multi-point coordinates=\\"35.7040744,139.5577317\\"></ol-geom-multi-point>
        <ol-style>
          <ol-style-circle radius=\\"10\\">
            <ol-style-fill color=\\"white\\"></ol-style-fill>
            <ol-style-stroke color=\\"blue\\" width=\\"4\\"></ol-style-stroke>
          </ol-style-circle>
        </ol-style>
      </ol-feature>
    </ol-source-vector>
  </ol-vector-layer>
</ol-map>"
`;

exports[`OpenLayerMapPointViewer > Render correctly 1`] = `
"<ol-map load-tiles-while-animating=\\"true\\" load-tiles-while-interacting=\\"true\\" style=\\"height: 400px;\\">
  <ol-view rotation=\\"0\\" zoom=\\"17\\" projection=\\"EPSG:4326\\"></ol-view>
  <ol-tile-layer>
    <ol-source-osm></ol-source-osm>
  </ol-tile-layer>
  <ol-vector-layer>
    <ol-source-vector>
      <ol-feature>
        <ol-geom-multi-point></ol-geom-multi-point>
        <ol-style>
          <ol-style-circle radius=\\"10\\">
            <ol-style-fill color=\\"white\\"></ol-style-fill>
            <ol-style-stroke color=\\"blue\\" width=\\"4\\"></ol-style-stroke>
          </ol-style-circle>
        </ol-style>
      </ol-feature>
    </ol-source-vector>
  </ol-vector-layer>
</ol-map>"
`;
